web/server/codechecker_server/session_manager.py

```diff
@@ -480,6 +495,21 @@ def __update_groups(self, user_name, groups):
 
         return False
 
+    def __try_regex_groups(self, username):
+        """
+        Return a set of groups that the user belongs to, depending on whether
+        the username matches the regular expression of the group.
+
+        """
+        matching_groups = set()
+        if self.__regex_groups_enabled:
+            for group_name, regex_list \
+                    in self.__group_regexes_compiled.items():
+                for r in regex_list:
+                    if re.search(r, username):
+                        matching_groups.add(group_name)
+        return matching_groups
```

Comment:
Just to simply the code we can save 1 indentation with this trick:
```suggestion
        matching_groups = set()
        if not self.__regex_groups_enabled:
            return matching_groups
 
        for group_name, regex_list \
                in self.__group_regexes_compiled.items():
            for r in regex_list:
                if re.search(r, username):
                    matching_groups.add(group_name)

        return matching_groups
```